statsmodels/multivariate/factor.py

```diff
@@ -0,0 +1,330 @@
+# -*- coding: utf-8 -*-
+
+from statsmodels.base.model import Model
+from .factor_rotation import rotate_factors, promax
+
+try:
+    import matplotlib.pyplot as plt
+    missing_matplotlib = False
+except ImportError:
+    missing_matplotlib = True
+
+if not missing_matplotlib:
+    from .plots import plot_scree, plot_loadings
+
+import numpy as np
+from numpy.linalg import eigh, inv, norm, matrix_rank
+import pandas as pd
+
+from statsmodels.iolib import summary2
+
+
+class Factor(Model):
+    """
+    Factor analysis
+
+    .. [1] Hofacker, C. (2004). Exploratory Factor Analysis, Mathematical Marketing.
+    http://www.openaccesstexts.org/pdf/Quant_Chapter_11_efa.pdf
+
+    Supported rotations:
+        'varimax', 'quartimax', 'biquartimax', 'equamax', 'oblimin',
+        'parsimax', 'parsimony', 'biquartimin', 'promax'
+
+    Parameters
+    ----------
+    endog : array-like
+        Variables in columns, observations in rows
+        Could be `None` if `corr` is not `None`
+    n_factor : int
+        The number of factors to extract
+    corr : array-like
+        Directly specify the correlation matrix instead of estimating from endog
+        If not `None`, `endog` will not be used
+    method : str
+        Specify the method to extract factors
+        'pa' - Principal axis factor analysis
+    smc : True or False
+        Whether or not to apply squared multiple correlations
+    endog_names: str
+        Names of endogeous variables.
+        If specified, it will be used instead of the column names in endog
+
+    """
+    def __init__(self, endog, n_factor, corr=None, method='pa', smc=True,
+                 missing='drop', endog_names=None):
```

Comment:
add also `nobs` as Kerby pointed out. 
It's not currently used but will be necessary in future when we get some inferential methods.